File: Codeforces/div805/C.cpp

#include <iostream>
#include <map>
#include <vector>

using namespace::std;

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		int n, k;
		cin >> n >> k;
		vector<int> a(n);
		for (int i=0; i < n; ++i)
			cin >> a[i];
		map<int, int> first;
		map<int, int> last;
		for (int i=0; i < n; ++i)
		{
			if (not first.contains(a[i]))
				first[a[i]] = i;
			last[a[i]] = i;
		}

		for (int i=0; i < k; ++i)
		{
			int aa, bb;
			cin >> aa >> bb;
			if (not first.contains(aa) or not first.contains(bb))
				cout << "NO" << endl;
			else if (first[aa] < last[bb])
				cout << "YES" << endl;
			else
				cout << "NO" << endl;
		}
	}
}

File: Codeforces/div805/C.py

def inp():
    return list(map(int, input().split()))


for _ in range(int(input())):
    input()
    n, k = inp()
    a = inp()
    first = dict()
    last = dict()
    for i in range(n):
        if a[i] not in first:
            first[a[i]] = i
        last[a[i]] = i

    for i in range(k):
        a, b = inp()
        if a not in first or b not in first:
            print('NO')
        elif first[a] < last[b]:
            print('YES')
        else:
            print('NO')

